### PR TITLE
update m5stack boards define

### PR DESCRIPTION
**Pull request by Gitshaoxiang**
## Description of Change

For better maintenance, we have made some changes
Unified the naming of M5Stack boards, modified some board default partition table options(solved some example app size too big),  also enabled the S3 board default CDC option, and added some new board options and custom partition tables.  

## Tests scenarios

These changes were tested on our official board management
`https://m5stack.oss-cn-shenzhen.aliyuncs.com/resource/arduino/package_m5stack_index_dev.json`

### CODE REVIEW

File: variants/m5stack_atom/pins_arduino.h [hunk 1]
```diff
@@ -3,15 +3,21 @@
 
 #include <stdint.h>
 
+#define EXTERNAL_NUM_INTERRUPTS 16
+#define NUM_DIGITAL_PINS        40
+#define NUM_ANALOG_INPUTS       16
+
+#define analogInputToDigitalPin(p)  (((p)<20)?(esp32_adc2gpio[(p)]):-1)
+#define digitalPinToInterrupt(p)    (((p)<40)?(p):-1)
+#define digitalPinHasPWM(p)         (p < 34)
```
Comment (P-R-O-C-H-Y) [inline variants/m5stack_atom/pins_arduino.h:12]: Please remove this part from all pins_arduino.h files of your boards.
These #defines are now defined globally per SoC.

Comment (Gitshaoxiang) [reply variants/m5stack_atom/pins_arduino.h]: Thanks for your reply, new changes have been committed now

**Comment by me-no-dev** [comment]
@P-R-O-C-H-Y PTAL

File: boards.txt [hunk 2]
```diff
@@ -15638,1365 +15638,4156 @@ espino32.menu.EraseFlash.all.upload.erase_cmd=-e
...
+m5stack_core.serial.disableDTR=true
+m5stack_core.serial.disableRTS=true
+
+m5stack_core.build.tarch=xtensa
+m5stack_core.build.bootloader_addr=0x1000
+m5stack_core.build.target=esp32
+m5stack_core.build.mcu=esp32
+m5stack_core.build.core=esp32
+m5stack_core.build.variant=m5stack_core
+m5stack_core.build.board=m5stack_core
```
Comment (P-R-O-C-H-Y) [inline boards.txt:15664]: build.board must be uppercase.
```suggestion
m5stack_core.build.board=M5STACK_CORE
```

**Review by P-R-O-C-H-Y — CHANGES_REQUESTED** [review]
@Gitshaoxiang The `<board>.build.board=` parameter must be uppercase. Please change it in all your boards.

File: boards.txt (hunk 2 above)
Comment (Gitshaoxiang) [reply boards.txt:15664]: updated

**Review by P-R-O-C-H-Y — APPROVED** [review]
Looks good :) LGTM

**Pull request merged**